Automattic/jetpack
# Plugin Scheduler: Add IsManaged key to plugins list

Fixes https://github.com/Automattic/dotcom-forge/issues/5665

## Proposed changes:

This adds a key `isManaged` to the return value of `get_plugins()` which is then synced via Jetpack Sync to WP.com's cache site. isManaged is true if the plugin is managed on Atomic, i.e. it is symlinked and its target points to a `wordpress` directory. This is used by scheduled updates to understand if the plugin can be updated or not.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Follow PCYsg-Osp-p2 to set up a testing environment on a WoA dev site.
* Use the Jetpack Beta plugin to activate WordPress.com Features for this branch.
* Refresh the installed plugins page on the site.
* Now apply D139262-code on WP.com.
* Go to the developer console and test the endpoint GET v1.2 `/sites/woawpcomstaging.com/plugins` on your WoA site. 
* You should see `is_managed` `true` and `false` correctly.

![CleanShot 2024-02-27 at 14 21 25](https://github.com/Automattic/jetpack/assets/533/c613b585-f106-4a67-ae1c-15c59a472c8f)
